compiler/src/iree/compiler/GlobalOptimization/Utils.cpp

```diff
@@ -99,6 +101,28 @@ Value createGenericElementwiseCastOp(
       .getResult(0);
 }
 
+FailureOr<IREE::Flow::DispatchRegionOp>
+wrapConsecutiveOpsInDispatchRegion(RewriterBase &rewriter,
+                                   SmallVector<Operation *> ops) {
+  FailureOr<IREE::Flow::DispatchRegionOp> maybeRegionOp =
+      IREE::Flow::wrapOpInDispatchRegion(rewriter, ops.back());
+  if (failed(maybeRegionOp)) {
+    return failure();
+  }
+  IREE::Flow::DispatchRegionOp regionOp = maybeRegionOp.value();
+
+  SmallVector<Operation *> precedingOps(ops.begin(), ops.end() - 1);
+  FailureOr<IREE::Flow::DispatchRegionOp> maybeFusedRegionOp =
+      IREE::Flow::movePrecedingOpsIntoDispatchRegion(rewriter, precedingOps,
+                                                     regionOp);
+  if (failed(maybeFusedRegionOp)) {
+    return failure();
+  }
+  regionOp = maybeFusedRegionOp.value();
+
+  return regionOp;
```

Comment:
why not directly return the result from `movePrecedingOpsIntoDispatchRegion`?

```suggestion
  return IREE::Flow::movePrecedingOpsIntoDispatchRegion(rewriter, precedingOps,
                                                        regionOp);
```